back/src/Guide/Types/Core.hs

```diff
@@ -74,6 +76,20 @@ instance Aeson.ToJSON Trait where
   toJSON = Aeson.genericToJSON Aeson.defaultOptions {
     Aeson.fieldLabelModifier = over _head toLower . drop (T.length "trait") }
 
+instance Aeson.FromJSON Trait where
```

Comment:
If this instance is not using `generic`, the `ToJSON` instance should also not use `generic`.